### PR TITLE
DOC: Mention that ``sparse.bsr_array`` does not support slicing.

### LINKED ISSUE #19042: DOC: Scipy Sparse BSR does not support slicing: __getitem__ not implemented
### Issue with current documentation:

The documentation for the BSR (Block Compressed Row) at https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.bsr_array.html#scipy.sparse.bsr_array does not mention that slicing, e.g., a[0:3, 1:4] is not supported for this sparse matrix sub-class.

It took me a while to find in the source code for `_bsr.py` that the `__getitem__` function raises a `NotImplementedError` and to figure out that this is the only sparse matrix sub-class that does this (the rest default to the `__getitem__` method in `_index.py`.

Please add this to the documentation, along with an explanation as to why this has not been implemented.

### Idea or request for content:

Right after the following text in the class bsr_matrix comment string:

    Notes
    -----
    Sparse matrices can be used in arithmetic operations: they support
    addition, subtraction, multiplication, division, and matrix power.

Add the following text.

    The BSR format does not support slicing, e.g., m[0:1, 3:4] and is the only sparse matrix
     type to not support this.  Please use or convert to a different sparse array format if
     you need this functionality.

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by m-maggi**
#### Reference issue
Attempt to close #19042

#### What does this implement/fix?
Further elaborate in the documentation on the slicing possibilities on sparse arrays.

### CODE REVIEW

**Review by j-bowhay — CHANGES_REQUESTED** [review]
Thanks @m-maggi! Can you add this to the `bsr_array` docstring too please

**Review by lucascolley — DISMISSED** [review]
also see the comments in the linked issue from @dschult

**Comment by m-maggi** [comment]
@j-bowhay @lucascolley I updated the docstrings of the objects not supporting the slicing following the comment of @dschult

File: scipy/sparse/__init__.py [hunk 1]
```diff
@@ -32,6 +32,8 @@
      that all *slicing* operations on these objects must produce
      two-dimensional results, or they will result in an error. This
      will be addressed in a future version.
+     Currently *slicing* is not supported by `bsr_array`, `coo_array`,
+     `dia_array`.
```
Comment (dschult) [inline scipy/sparse/__init__.py:36]: I don't believe this note belongs in this place in the docs. It is not a difference between array and matrix interfaces. And we don't include a description of all the many differences in support between the formats here. 

So, I would prefer we remove this note, and focus on the doc_strings for the individual formats.

File: scipy/sparse/_bsr.py [hunk 2]
```diff
@@ -703,6 +703,7 @@ class bsr_array(_bsr_base, sparray):
     -----
     Sparse arrays can be used in arithmetic operations: they support
     addition, subtraction, multiplication, division, and matrix power.
+    Block Sparse Row format sparse arrays do not support slicing.
```
Comment (dschult) [inline scipy/sparse/_bsr.py:706]: Can we move this to the bottom of the Notes section with a heading of `**Limitations**`?

Similar changes to `bsr_matrix`.

File: scipy/sparse/_coo.py [hunk 3]
```diff
@@ -699,6 +699,7 @@ class coo_array(_coo_base, sparray):
 
     Sparse arrays can be used in arithmetic operations: they support
     addition, subtraction, multiplication, division, and matrix power.
+    Sparse arrays in COOrdinate format do not support slicing.
```
Comment (dschult) [inline scipy/sparse/_coo.py:702]: The COO doc_string already lists a disadvantage being that it does not support slicing. Maybe we don't need anything further?

File: scipy/sparse/_dia.py [hunk 4]
```diff
@@ -481,6 +481,7 @@ class dia_array(_dia_base, sparray):
 
     Sparse arrays can be used in arithmetic operations: they support
     addition, subtraction, multiplication, division, and matrix power.
+    Sparse arrays with DIAgonal storage do not support slicing.
```
Comment (dschult) [inline scipy/sparse/_dia.py:484]: This DIA form looks good to me! :)

**Review by dschult — COMMENTED** [review]
Thanks for taking this on. See specific comments below. I think the comments should appear in the class doc_strings. And it seems the format of those doc_strings vary enough that a standard pattern may not match all three formats.

Thanks!

**Comment by m-maggi** [comment]
thank you for your comments @dschult . How does it look like now?

File: scipy/sparse/_coo.py [hunk 5]
```diff
@@ -813,7 +813,7 @@ class coo_matrix(spmatrix, _coo_base):
 
     Sparse matrices can be used in arithmetic operations: they support
     addition, subtraction, multiplication, division, and matrix power.
-
+
```
Comment (dschult) [inline scipy/sparse/_coo.py:816]: Can you take out the 4 white spaces on this line?
Then I think it is ready.
Thanks!

Comment (m-maggi) [reply scipy/sparse/_coo.py]: Thanks for spotting this, now _coo.py should have no differences compared to the main branch.

**Review by dschult — APPROVED** [review]

**Review by j-bowhay — APPROVED** [review]
Thanks everyone

**Pull request merged**